src/test/java/com/redhat/cloud/event/parser/modules/DateTimeModuleTest.java

```diff
@@ -97,4 +97,16 @@ public void withNegativeOffsetToReachNextDay() throws JsonProcessingException {
         assertEquals(LocalDateTime.of(2021, 3, 13, 18, 44, 0), helper.localDateTime);
         assertEquals(OffsetDateTime.of(2021, 3, 14, 0, 44, 0, 0, ZoneOffset.UTC), helper.offsetDateTime);
     }
+
+    @Test
+    public void withNanoseconds() throws JsonProcessingException {
+        Helper helper = new Helper();
+        helper.localDateTime = LocalDateTime.parse("2023-05-10T06:02:53.704467873");
+
+        String serialized = objectMapper.writeValueAsString(helper);
+
+        assertEquals("{\"localDateTime\":\"2023-05-10T06:02:53.704467873Z\",\"offsetDateTime\":null}", serialized);
+
+        assertEquals(LocalDateTime.of(2023, 5, 10, 6, 2, 53, 704467873), helper.localDateTime);
+    }
```

Comment:
Here you are testing that a LocalDateTime can be serialized - not the other way around which was failing (IIUC)